src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -206,7 +208,14 @@ public String getId() throws AbortException {
     public @Nonnull Map<String,String> getBuildVariables() throws AbortException {
         Run<?,?> build = build();
         if (build instanceof AbstractBuild) {
-            return Collections.unmodifiableMap(((AbstractBuild<?,?>) build).getBuildVariables());
+            Map<String,String> buildVars = new HashMap<>();
+            try {
+                buildVars.putAll(build.getEnvironment(TaskListener.NULL));
+            } catch (IOException | InterruptedException e) {
+                // Do nothing
```

Comment:
Should be logged. Or simply thrown up out of this method so it shows up as a script failure to be fixed.